dnsmasq_interface.c

```diff
@@ -167,7 +150,12 @@ void FTL_new_query(unsigned int flags, char *name, struct all_addr *addr, char *
 	queries[queryID].db = false;
 	queries[queryID].id = id;
 	queries[queryID].complete = false;
-	queries[queryID].private = (config.privacylevel == PRIVACY_MAXIMUM);
+
+	// Check and apply possible privacy level rules
+	// We do this immediately on the raw data to avoid any possible leaking
```

Comment:
The second line of this comment may need to be tweaked, since no data is modified by getting the privacy level. Instead, the current privacy level is remembered for that query.